src/state_machine.zig

```diff
@@ -1383,31 +1211,40 @@ pub fn StateMachineType(
     };
 }
 
-fn sum_overflows(a: u64, b: u64) bool {
-    _ = std.math.add(u64, a, b) catch return true;
+fn sum_overflows(comptime Int: type, a: Int, b: Int) bool {
+    comptime assert(Int != comptime_int);
+    comptime assert(Int != comptime_float);
+    _ = std.math.add(Int, a, b) catch return true;
     return false;
 }
 
 /// Optimizes for the common case, where the array is zeroed. Completely branchless.
-fn zeroed_16_bytes(a: [16]u8) bool {
-    const x = @as([2]u64, @bitCast(a));
+inline fn zeroed_16_bytes(a: [16]u8) bool {
+    const x: [2]u64 = @bitCast(a);
     return (x[0] | x[1]) == 0;
 }
 
-fn zeroed_32_bytes(a: [32]u8) bool {
-    const x = @as([4]u64, @bitCast(a));
+inline fn zeroed_32_bytes(a: [32]u8) bool {
+    const x: [4]u64 = @bitCast(a);
     return (x[0] | x[1] | x[2] | x[3]) == 0;
 }
 
-fn zeroed_48_bytes(a: [48]u8) bool {
-    const x = @as([6]u64, @bitCast(a));
+inline fn zeroed_48_bytes(a: [48]u8) bool {
+    const x: [6]u64 = @bitCast(a);
     return (x[0] | x[1] | x[2] | x[3] | x[4] | x[5]) == 0;
 }
 
+/// Optimizes for the common case, where the arrays are equal. Completely branchless.
+inline fn equal_4_bytes(a: [4]u8, b: [4]u8) bool {
```

Comment:
It looks like these `equal_*_bytes` and `zeroed_*_bytes` aren't actually used; can they be removed?